### ISSUE TITLE
Auto DIM generator

### ISSUE THREAD

**Issue by Nextric** [gusmanb/ZXBasicStudio#36]
Mmm, I think it could be very useful to autogenerate the array structures, since when they are very large it is very tedious to write them. I don't know if with a separate tool or with a context menu when writing the DIM.

**Comment by gusmanb** [comment]
Can you explain a bit more? What do you mean?

**Comment by Nextric** [comment]
Of Course. For example, when I define this array. DIM (3,8) => , to autogenerate the bracket structure empty.
{ _ 
    { , , , , , , , }, _
    {  , , , , , , , }, _
    {  , , , , , , , } _
}

**Comment by SirRickster** [comment]
Tools
--- Code Snippets
------- Dim Generator
------- xxxxxx
------- xxxxxx

it could be a good idea

**Comment by gusmanb** [comment]
Ok, now the final question :D

Is this really useful? I mean, suppose that it writes the structure of the arrays, you are going to have to go to each comma, insert a value, jump the comma, add the next value, repeat until line ends, change line and so on... At the end you are going to use the same ammount of keystrokes unless you use the mouse and that's even worse.

Maybe this could go in a different way. Suppose that when the IDE detects that you are creating and initializing an array and you press crtl+space (like with autocomplete request) a pop-up opens, the pop-up contains a grid with textboxes mimicking the distribution of the array, then creating the array would be "enter number" "press tab" and repeat until all the textboxes are filled, you accept and it writes the array to the text editor fully populated.

This would be limited to unidimensional and bidimensional arrays, and there must be also a limit to the ammount of total items as it could be very heavy (imagine that you define an array of 4096 elements... xD).

What do you think?

**Comment by Nextric** [comment]
Wow! Your idea is bether than of mine. I see perfect!
